packages/flutter/lib/src/cupertino/form_row.dart

```diff
@@ -0,0 +1,198 @@
+/// [prefix] and [child]. When a [Text] is given to [error], it will be shown in
+/// [CupertinoColors.destructiveRed] coloring and medium-weighted font.
+///
+/// {@tool snippet}
+///
+/// Creates a [CupertinoFormSection] containing a [CupertinoFormRow] with the
+/// [prefix], [child], [helper] and [error] widgets.
+///
+/// ```dart
+/// class FlutterDemo extends StatefulWidget {
+///   FlutterDemo({Key key}) : super(key: key);
+///
+///   @override
+///   _FlutterDemoState createState() => _FlutterDemoState();
+/// }
+///
+/// class _FlutterDemoState extends State<FlutterDemo> {
+///   bool toggleValue = false;
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     return CupertinoPageScaffold(
+///       child: Center(
+///         child: CupertinoFormSection(
+///           title: "Section 1",
+///           children: [
+///             CupertinoFormRow(
+///               prefix: Text('Toggle'),
+///               helper: Text('Use your instincts'),
+///               error: toggleValue ? Text('Can't be true') : null,
+///               child: CupertinoSwitch(
+///                 value: this.toggleValue,
+///                 onChanged: (value) {
+///                   setState(() {
+///                     this.toggleValue = value;
+///                   });
+///                 },
+///               ),
+///             ),
+///           ],
+///         ),
+///       ),
+///     );
+///   }
+/// }
+/// ```
+/// {@end-tool}
+class CupertinoFormRow extends StatelessWidget {
+  /// Creates an iOS-style split form row with a standard prefix and child widget.
+  /// Also provides a space for error and helper widgets that appear underneath.
+  ///
+  /// The [child] parameter is required. This widget is displayed at the end of
+  /// the row.
+  ///
+  /// The [prefix] parameter is optional and is displayed at the start of the
+  /// row. Standard iOS guidelines encourage passing a [Text] widget to [prefix]
+  /// to detail the nature of the row's [child] widget.
+  ///
+  /// The [margins] parameter is used to pad the contents of the row. It defaults
+  /// to the standard iOS padding. If no edge insets are intended, explicitly
+  /// pass [EdgeInsets.zero] to [margins].
+  ///
+  /// The [helper] and [error] parameters are both widgets targeted at displaying
+  /// more information about the row. Both widgets are placed underneath the
+  /// [prefix] and [child]. When a [Text] is given to [error], it will be shown in
+  /// [CupertinoColors.destructiveRed] coloring and medium-weighted font.
+  const CupertinoFormRow({
+    Key? key,
+    required this.child,
+    this.prefix,
+    this.margins,
+    this.helper,
+    this.error,
+  }) : super(key: key);
+
+  /// A widget that is displayed at the start of the row.
+  ///
+  /// The [prefix] parameter is displayed at the start of the row. Standard iOS
+  /// guidelines encourage passing a [Text] widget to [prefix] to detail the
+  /// nature of the row's [child] widget.
```

Comment:
Generally, for field docs, describe if it can be null and what happens if it is null. 

Context: these doc text appear in the IDE hovers and in the API doc site (https://api.flutter.dev/flutter/cupertino/CupertinoNavigationBar/middle.html) on their own without much context for the class so they should be as self-descriptive as possible.